engine/base.ftl

```diff
@@ -1470,3 +1470,303 @@ A comparator is an operator and a version
+value of the entity based on the CombineBehaviour as defined by the
+combineEntities() base function.
+
+More than one qualifier may match, in which case the qualifiers are applied to
+the nominal value in the order in which the qualifiers are defined.
+
+One or more qualifiers can be added to any entity via a reserved "Qualifiers"
+attribute. Where the entity to be qualified is not itself an object, the
+desired entity must be wrapped in an object in order that the "Qualifiers" attribute
+can be attached. Note thet the  type of the result will be that of the provided
+value.
+
+There is a long form and a short form value for Qualifiers.
+
+In the long form, the "Qualifiers" attribute value is a list of qualifier objects.
+
+Each qualifier object must have a "Filter" attribute and a "Value" attribute, as well
+as optional "MatchBehaviour" and "DefaultBehaviour" attributes. By default, the
+MatchBehaviour is ONETOONE_FILTER_MATCH_BEHAVIOUR and the Combine Behaviour is
+MERGE_COMBINE_BEHAVIOUR.
+
+The long form gives full control over the qualification process and the order
+in which qualifiers are applied. In the following example, the nominal value is 100,
+but 50 will be used assuming the Environment is prod;
+
+{
+    "Value" : 100,
+    "Qualifiers : [
+        {
+            "Filter" : {"Environment" : "prod"},
+            "MatchBehaviour" : ONETOONE_FILTER_MATCH_BEHAVIOUR,
+            "Value" : 50,
+            "CombineBehaviour" : MERGE_COMBINE_BEHAVIOUR
+        }
+    ]
+}
+
+In the short form, the "Qualifiers" attribute value is an object.
+
+Each attribute of the attribute value represents a qualifier. The attribute key
+is the value of  the "Any" attribute of the Match Filter, the MatchBehaviour is
+ANY_FILTER_MATCH_BEHAVIOUR, and the CombineBehaviour is MERGE_COMBINE_BEHAVIOUR.
+The attribute value is the value of the attribute.
+
+Because object attribute processing is not ordered, the short form does not provide fine
+control in the situation where multiple qualifiers match - effectively they need
+to be independent. For consistency, attributes are sorted alphabetically before processing.
+
+The short form is useful for simple situations such as setting variation based
+on environment. Assuming environment values are unique, the long form example
+could be simplified to
+
+{
+  "Value" : 100,
+  "Qualifiers" : { "prod" : 50}
+}
+
+Equally
+
+{
+  "Value" : 100,
+  "Qualifiers" : { "prod" : 50, "industry" : 23}
+}
+
+is the equivalent of
+
+{
+    "Value" : 100,
+    "Qualifiers : [
+        {
+            "Filter" : {"Any" : "prod"},
+            "MatchBehaviour" : ANY_FILTER_MATCH_BEHAVIOUR,
+            "Value" : 50,
+            "CombineBehaviour" : MERGE_COMBINE_BEHAVIOUR
+        },
+        {
+            "Filter" : {"Any" : "industry"},
+            "MatchBehaviour" : ANY_FILTER_MATCH_BEHAVIOUR,
+            "Value" : 23,
+            "CombineBehaviour" : MERGE_COMBINE_BEHAVIOUR
+        }
+    ]
+}
+
```

Comment:
Can I use both approaches? The second approach while more verbose is far easier to understand. Looking at the qualifiers in the simplified form is really hard to follow. 

What is prod? what is industry? How as a user can I find out what is available for qualification?

Reply:
Any provides a real potential for danger or misunderstanding I think

Reply:
Only one approach per `Qualifiers` attribute as the content (hash or sequence) then drives what structure is expected.

any is what we have now - we can transition away from it but it is there to supported existing qualification in cmdbs.